### PR TITLE
Merge 1.17.1 into 1.18.2

### PR DESCRIPTION
**Pull request by ZacSharp**
As always, if you update 1.17.1 I'd like to remerge before this one is merged.

### CODE REVIEW

File: src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java [hunk 1]
```diff
@@ -65,6 +66,27 @@ public IStaticSchematic parse(InputStream input) throws IOException {
...
+            int version = nbt.getInt("Version");
+            switch (version) {
+                case 4: //1.12
+                    throw new UnsupportedOperationException("This litematic Version is too old.");
+                case 5: //1.13-1.17
+                    return new LitematicaSchematic(nbt, false);
+                case 6: //1.18+
+                    throw new UnsupportedOperationException("This litematic Version is too new.");
+                default:
+                    throw new UnsupportedOperationException("Unsuported Version of a Litematica Schematic");
```
Comment (rycbar0) [inline src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java:87]:
```
switch (version) {
 case 4: //1.12
 case 5: //1.13-1.17
  throw new UnsupportedOperationException("This litematic Version is too old.");
 case 6:
  return new LitematicaSchematic(nbt, false);
 default:
  throw new UnsupportedOperationException("Unsuported Version of a Litematica Schematic");
```

**Review by rycbar0 — CHANGES_REQUESTED** [review]
the switch statement in the litematic version check needs to be adjusted

File: src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java (hunk 1 above)
Comment (wagyourtail) [reply src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java:87]: do you have to discard older versions? or is there a way to work with them?

Comment (rycbar0) [reply src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java:87]: there are blocks that will work fine like minecraft:glass over all versions.
other blocks like oak planks will be replaced with air blocks when they not match pre or post flattening.
there shouldnt be any issues between version 5 and 6 but litematica itself does deny loading a schematic from a version other then its own.
in short it wont break anything if we dont test the version but the shematic will maybe be built wrong/incompleet

Comment (ZacSharp) [reply src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java:87]: Are there any differences in parsing?

Comment (rycbar0) [reply src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java:87]: in the parsing itself no.
but a pre flattening 1.12 schematic with minecraft:stone{variant: garnite} cant be resolved in a post flattening verison and defaults to a minecraft:air block and vice versa

**Pull request merged**